### PR TITLE
Improve performance by using numpy for trigger detection and closing matplotlib figures to avoid memleaks.

### PR DESCRIPTION
**Pull request by bpinsard**
cutting acq files into runs lead to very unreasonably large memory usage, and long runtime.
After memory and time profiling here are the proposed changes.
I ran the test locally and some crashed but do not seem related to my changes. I fixed one of the fail test by adding the missing requirements. 

## Proposed Changes

  - properly close matplotlib figures after rendering/saving to avoid clogging memory
  - replace loop-based trigger counting with a more (100-fold) efficient numpy-based one. 
  - fix requirements to have some failing test passing

## Change Type

- [ ] `bugfix` (+0.0.1)
- [x] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `refactoring` (no version update)
- [ ] `test` (no version update)
- [ ] `infrastructure` (no version update)
- [ ] `documentation` (no version update)
- [ ] `other`

## Checklist before review

- [ ] I added everything I wanted to add to this PR.
- [ ] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [x] \[Code or tests only\] I ran and passed tests locally. 
- [ ] \[Documentation only\] I built the docs locally.
- [x] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [ ] My code respects the adopted style, especially linting conventions.
- [ ] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [ ] I added or indicated the right labels.

- [ ] I added information regarding the timeline of completion for this PR.

- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!

### CODE REVIEW

File: phys2bids/physio_obj.py
Comment (smoia) [inline phys2bids/physio_obj.py:527]:
```suggestion
```

File: phys2bids/physio_obj.py
Comment (smoia) [inline phys2bids/physio_obj.py:528]: This does look like a better alternative, thank you for that!
We've been refactoring parts of the code recently to make them more numpy-based and perform better, so it's a very well appreciated change.

File: phys2bids/viz.py
Comment (smoia) [inline phys2bids/viz.py:270]: I'm not sure the clf is necessary if you are closing the figures. But I need to double check myself.
```suggestion
```

File: phys2bids/viz.py
Comment (smoia) [inline phys2bids/viz.py:271]: Could you add this line in all the other plots as well, please?

File: requirements.txt
Comment (smoia) [inline requirements.txt:4]:
```suggestion
```

`pymatreader` is not a mandatory requirement: `phys2bids` installation is modular, so if you want to be able to read matlab file, you need to install `pymatreader`, if not you don't. same thing for Acqknowledge or spike2 files. Similarly, your tests might be failing because you'd need to install physbids with either the `dev` or the `all` option
(see [here](https://phys2bids.readthedocs.io/en/latest/contributing.html#full-developer-installation)).

It's true that we did not update the documentation accordingly, so thank you for pointing it out! I will open an issue about it.

**Review by smoia — CHANGES_REQUESTED** [review]
Hello @bpinsard, thank you for this PR!
There's a couple of comments scattered around, but it's quick stuff!

**Comment by smoia** [comment]
All of the tests passed remotely (although I'm missing pre-commit, so let me check that), so I'm fairly sure we're good to go once the small changes have been incorporated!

**Comment by smoia** [comment]
pre-commit found ;)
One last thing before merging: we care a lot about PR titles, since they are automatically used to compile the changelog.

Can you edit the title to make it more self-explanatory (yet concise)?
I started a hint, but feel free to change it completely!

**Comment by bpinsard** [comment]
Added testing for the memory issue.

**Comment by smoia** [comment]
:rocket: PR was released in [`2.10.0`](https://github.com/physiopy/phys2bids/releases/tag/2.10.0) :rocket:

**Comment by bpinsard** [comment]
Thanks for your fast processing of the PR!

**Pull request merged**
